### PR TITLE
Cloud tagging based cleanup

### PR DESCRIPTION
**Pull request by fruch**

### CODE REVIEW

**Comment by fruch** [comment]
this should fix #781

File: sdcm/utils.py [hunk 1]
```diff
@@ -190,3 +195,57 @@ def upload_file(cls, file_path):
         except Exception as e:
             logger.debug("Unable to upload to S3: %s" % e)
             return ""
+
+
+aws_regions = ['us-east-1', 'eu-west-1']
+gce_regions = ['us-east1-b', 'us-west1-b', 'us-east4-b']
+
+
+def clear_cloud_instances(tags_dict):
```
Comment (bentsi) [inline sdcm/utils.py]: `clear` -> `clean`

File: sdcm/utils.py [hunk 2]
```diff
@@ -190,3 +195,57 @@ def upload_file(cls, file_path):
...
+        driver = ComputeEngine(gcp_credentials["project_id"] + "@appspot.gserviceaccount.com",
+                               gcp_credentials["private_key"],
+                               datacenter=region,
+                               project=gcp_credentials["project_id"])
+
+        for node in driver.list_nodes():
+            tags = node.extra['metadata'].get('items', [])
+
+            if all([any([t['key'] == key and t['value'] == value for t in tags]) for key, value in tags_dict.items()]):
+                print node
```
Comment (bentsi) [inline sdcm/utils.py]: logger instead print

File: sdcm/utils.py [hunk 3]
```diff
@@ -190,3 +195,57 @@ def upload_file(cls, file_path):
...
+            for instance in response['TerminatingInstances']:
+                assert instance['CurrentState']['Name'] in ['terminated', 'shutting-down']
+
+
+def clear_instances_gce(tags_dict):
+    ks = KeyStore()
+    gcp_credentials = ks.get_gcp_credentials()
+    ComputeEngine = get_driver(Provider.GCE)
+    for region in gce_regions:
+        driver = ComputeEngine(gcp_credentials["project_id"] + "@appspot.gserviceaccount.com",
```
Comment (bentsi) [inline sdcm/utils.py]: add printout "cleaning region X"

File: sdcm/utils.py [hunk 4]
```diff
@@ -190,3 +195,57 @@ def upload_file(cls, file_path):
...
+                               gcp_credentials["private_key"],
+                               datacenter=region,
+                               project=gcp_credentials["project_id"])
+
+        for node in driver.list_nodes():
+            tags = node.extra['metadata'].get('items', [])
+
+            if all([any([t['key'] == key and t['value'] == value for t in tags]) for key, value in tags_dict.items()]):
```
Comment (bentsi) [inline sdcm/utils.py]: too complex, or simplify with ifs or add a comment what this part does

File: sdcm/cluster_gce.py [hunk 5]
```diff
@@ -45,8 +46,10 @@ def __init__(self, gce_instance, gce_service, credentials,
                           cluster.TEST_DURATION, cluster.Setup.KEEP_ALIVE)
             self._instance_wait_safe(self._gce_service.ex_set_node_tags,
                                      self._instance, ['keep-alive'])
-        self._instance_wait_safe(self._gce_service.ex_set_node_metadata,
-                                 self._instance, {'workspace': cluster.WORKSPACE, 'uname': ' | '.join(os.uname())})
+
+        # fruch: test if needed
+        # self._instance_wait_safe(self._gce_service.ex_set_node_metadata,
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster_gce.py]: no we don't need it

**Pull request merged**